src/pkgs/qemu-aarch64-static.nix

```diff
@@ -0,0 +1,16 @@
+{ buildPkgs }:
+
+buildPkgs.stdenv.mkDerivation {
+  name = "qemu-aarch64-static";
+
+  src = builtins.fetchurl {
+    url = "https://github.com/multiarch/qemu-user-static/releases/download/v4.1.0-1/qemu-aarch64-static";
```

Comment:
Actually, Nix-provided QEMU would be perfectly fine. I pulled this one in because it was easier than and allowed to build the zipball without Nix. Not there is little or no point in pulling in some random statically linked binary. Sorry for that.

Reply:
Sadly this does not work out of the box.. 

With `qemu-aarch64` I get
```
/nix/store/yswvqjahp9bqi17by579cs6q7i7qz7b5-qemu-3.1.0/bin/qemu-aarch64: error while loading shared libraries: libpcre.so.1: cannot open shared object file: No such file or directory
```

With `qemu-system-aarch64` I get
```
/nix/store/yswvqjahp9bqi17by579cs6q7i7qz7b5-qemu-3.1.0/bin/qemu-system-aarch64: error while loading shared libraries: libssl3.so: cannot open shared object file: No such file or directory
```

I really don't know what are the differences between `qemu-aarch64`, `qemu-system-aarch64` and `qemu-aarch64-static` and what tool is best suited for this case. If you prefer one, please let me know. Maybe I can fix the issues with these shared libraries.

Reply:
Aahh, yes, now I remember.

The difference, in short, goes like that:
* qemu-system-<arch> emulates the whole system: CPU, MMU, peripherals, you name it. It is usually used with kernels and rootfs images, or disk images that contain a bootloader + that.
* qemu-<arch> or qemu-user-<arch> emulates a CPU and executes a binary for a different architecture, passing the syscalls to the host OS. This has an effect of either transparently running statically-linked executables built for different arches -or- chrooting into root filesystems full of binaries and libraries, for, e.g., aarch64. This is a very common scenario in the embedded world, so common that proot has support for doing that plumbed in. Unfortunately,
* qemu-<arch>-static is required for that, because QEMU also needs libraries to work, and it's easier to just compile it statically.

Properly solving that would require building statically-linked QEMU with nix. I've filed #14 for that.